src/Psalm/Internal/BCHelper.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Psalm\Internal;
+
+use PhpParser;
+use PhpParser\Lexer\Emulative;
+use PhpParser\Node;
+use PhpParser\PhpVersion;
+
+use function class_exists;
+
+/**
+ * @internal
+ */
+final class BCHelper
+{
+    private const CLASS_MAP = [
+        Node\Expr\ArrayItem::class => Node\ArrayItem::class,
+        Node\Expr\ClosureUse::class => Node\ClosureUse::class,
+        Node\Scalar\DNumber::class => Node\Scalar\Float_::class,
+        Node\Scalar\Encapsed::class => Node\Scalar\InterpolatedString::class,
+        Node\Scalar\EncapsedStringPart::class => Node\InterpolatedStringPart::class,
+        Node\Scalar\LNumber::class => Node\Scalar\Int_::class,
+        Node\Stmt\DeclareDeclare::class => Node\DeclareItem::class,
+        Node\Stmt\PropertyProperty::class => Node\PropertyItem::class,
+        Node\Stmt\StaticVar::class => Node\StaticVar::class,
+        Node\Stmt\UseUse::class => Node\UseItem::class,
+    ];
+
+    public static function getPHPParserClassName(string $className): string
+    {
+        if (isset(self::CLASS_MAP[$className]) && class_exists(self::CLASS_MAP[$className])) {
+            return self::CLASS_MAP[$className];
+        }
+
+        return $className;
+    }
+
+    public static function usePHPParserV4(): bool
+    {
+        return class_exists('\PhpParser\Node\Stmt\Throw');
```

Comment:
This is missing an underscore in the class name.

```suggestion
        return class_exists('\PhpParser\Node\Stmt\Throw_');
```

Reply:
Thanks